girder/api/rest.py

```diff
@@ -965,7 +970,14 @@ def handleRoute(self, method, path, params):
             val = event.responses[0]
         else:
             self._defaultAccess(handler)
-            val = handler(**kwargs)
+
+            # Attempt to handle and trigger failed event if an
+            # exception is raised.
+            try:
+                val = handler(**kwargs)
+            except Exception as e:
+                events.trigger('.'.join((eventPrefix, 'failed')), kwargs)
+                raise e
```

Comment:
```suggestion
            except Exception:
                events.trigger('.'.join((eventPrefix, 'failed')), {'kwargs': kwargs})
                raise
```

Reply:
I didn't know you could re-raise like that, thanks!

What's the reason for wrapping `kwargs` in a dict? It seems to differ from how the `before` and `after` events include these.

Reply:
I don't have strong feelings, but passing it wrapped in an outer dict allows easily adding more info the event in the future, without breaking compatibility for anyone.

Reply:
I suppose that's true (I could image passing the exception itself as well) but I'd think consistency with the before and after events would be more important. Maybe not, @manthey any thoughts?

Reply:
I'd rather have the consistency.  To my knowledge, we haven't wanted to extend the before and after events.